### PR TITLE
Task4

### PR DESCRIPTION
**Pull request by kruchkou**

### CODE REVIEW

File: controller/src/main/java/com/epam/esm/controller/GiftCertificateController.java [hunk 1]
```diff
@@ -38,6 +40,7 @@ public GiftCertificateController(GiftCertificateService giftCertificateService,
     }
 
     @GetMapping("/{id}")
+    @PreAuthorize("hasRole('ANONYMOUS')")
```
Comment (SenPavel) [inline controller/src/main/java/com/epam/esm/controller/GiftCertificateController.java:43]: Is there a difference between @PreAuthorize("hasRole('ANONYMOUS')") and @PreAuthorize("isAnonymous()")?

File: repository/src/main/java/com/epam/esm/repository/OrderRepository.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+     */
+    Optional<Order> findById(int id);
+
+    /**
+     * Connects to database and returns Orders by user id
+     *
+     * @param id is Order ID value.
+     * @return List of all {@link Order} entities from database.
+     */
+    List<Order> findByUser_Id(int id);
```
Comment (SenPavel) [inline repository/src/main/java/com/epam/esm/repository/OrderRepository.java:28]: No need to use underscore

**Review by SenPavel — CHANGES_REQUESTED** [review]

File: controller/src/main/java/com/epam/esm/controller/GiftCertificateController.java (hunk 1 above)
Comment (kruchkou) [reply controller/src/main/java/com/epam/esm/controller/GiftCertificateController.java]: No difference, changed to isAnonymous()

File: repository/src/main/java/com/epam/esm/repository/OrderRepository.java (hunk 2 above)
Comment (kruchkou) [reply repository/src/main/java/com/epam/esm/repository/OrderRepository.java]: Fixed

**Review by SenPavel — APPROVED** [review]

**Pull request merged**